posts/deploying-a-jhipster-app-to-aws-using-elastic-beanstalk.md

```diff
@@ -0,0 +1,101 @@
+
+ The command `jhipster aws` will ask questions about the AWS deployment:
+
+ ![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/jhipster-aws.png)
+
+ The AWS account used needs access to the S3 bucket since it will be used as a storage location for the application's Jars. The sub-generator will also create a RDS instance so make sure that the database credentials are correct and match the ones in `application-prod.yml`. The application will always be packaged using the `prod` profile, which means that all the tests must be green!
+
+ Once the Elastic Beanstalk application is created, navigate to the [Elastic Beanstalk console](https://console.aws.amazon.com/elasticbeanstalk/home) to view the details of the application. Use the public URL to check that everything was correctly deployed and running.
+
+ ![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/eb-url.png)
+
+ If the deployment failed, the logs page can be used to view all the logs of the application. The logs from `/var/log/tomcat8/catalina.out` are the JHipster logs, those should be used to find out why the application did not start.
+
+# Elastic Beanstalk application improvements
+
+By default, your Elastic Beanstalk application uses a classic load balancer with auto scaling up to 4 instances. The default configuration is enough for demo purpose, but if you expect a heavy load I recommend updating the configuration. Also, it might be a better idea to set the scaling trigger to the `CPUUtilization` metric instead of `NetworkOut`.
+
+Another good practice is to enable `S3 log storage` or `CloudWatch Logs` in `Configuration > Software`. It is always a good idea to have a strong log retention policy with a production application in order to track faulty/suspicious behavior.
+
+ ![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/eb-configuration.png) 
+
+ The `Rolling updates and deployments` configuration should also be changed, especially if the application can't be out of service for few minutes. More details can be found on [the AWS documentation page](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/using-features.rolling-version-deploy.html?icmpid=docs_elasticbeanstalk_console).
+
+ # Conclusion
+
+Elastic Beanstalk saves you a lot of time and removes all the complexity of creating a production ready infrastructure from scratch. The JHipster AWS sub-generator is easy to use and will let you create your Elastic Beanstalk environment in few steps.
+
+Since the sub-generator uses a default Beanstalk configuration, I recommended tweaking it a little bit to fulfill your production requirements.
```

Comment:
In this conclusion, can you maybe explain why again we wanted this deployed via the Elastic Beanstalk service, how it was deployed using the sub-generator, and then the overview what the service setup?